modules/components/analytics/core/kpi/org.wso2.carbon.pc.analytics.core.kpi/src/main/java/org/wso2/carbon/pc/analytics/core/kpi/DASConfigClient.java

```diff
@@ -0,0 +1,173 @@
+                    dasPassword = secretResolver.resolve(AnalyticsConfigConstants.SECRET_ALIAS_DAS_PASSWORD)
+                            .toCharArray();
+                } else {
+                    dasPassword = DASConfigurationUtils.getDASPassword().toCharArray();
+                }
+            } else {
+                dasPassword = DASConfigurationUtils.getDASPassword().toCharArray();
+            }
+
+            dasUrl = DASConfigurationUtils.getDASURL();
+            dasConfigDetailsJOb = new JSONObject(dasConfigDetails);
+            streamName = dasConfigDetailsJOb.getString(AnalyticsConfigConstants.EVENT_STREAM_NAME);
+            stremaVersion = dasConfigDetailsJOb.getString(AnalyticsConfigConstants.EVENT_STREAM_VERSION);
+            streamId = dasConfigDetailsJOb.getString(AnalyticsConfigConstants.EVENT_STREAM_ID);
+            streamDescription = dasConfigDetailsJOb.getString(AnalyticsConfigConstants.EVENT_STREAM_DESCRIPTION);
+            streamNickName = dasConfigDetailsJOb.getString(AnalyticsConfigConstants.EVENT_STREAM_NICK_NAME);
+            receiverName = dasConfigDetailsJOb.getString(AnalyticsConfigConstants.EVENT_RECEIVER_NAME);
+            processVariables = dasConfigDetailsJOb.getJSONArray(AnalyticsConfigConstants.PROCESS_VARIABLES);
+
+            //login to DAS
+            loginServiceClient = PCAnalyticsServerHolder.getInstance().getLoginAdminServiceClient();
+            String session = loginServiceClient.authenticate(dasUsername, dasPassword);
+
+            //remove the password from memory
+            Arrays.fill(dasPassword, ' ');
+            dasPassword = null;
+
+            //create event stream // The payload data is as>> "process instance id, valueAvailability, actual process
+            // variables list
+            PCAnalyticsServerHolder.getInstance().getStreamAdminServiceClient()
+                    .createEventStream(session, streamName, stremaVersion, streamId, streamNickName, streamDescription,
+                            processVariables);
+            if (log.isDebugEnabled()) {
+                log.debug("Created the Event Stream: " + streamId + " in WSO2 DAS on :" + dasUrl);
+            }
+
+            //create event receiver
+            PCAnalyticsServerHolder.getInstance().getReceiverAdminServiceClient()
+                    .deployEventReceiverConfiguration(session, receiverName, streamId,
+                            AnalyticsConfigConstants.WSO2_EVENT);
+            if (log.isDebugEnabled()) {
+                log.debug("Created the Event Receiver: " + receiverName + " for the " + streamId + " in WSO2 DAS");
+            }
+
+            //now send the REST Post call to the WSO2 BPS to communicate the analytics configuration details to the
+            // BPS from PC
+            BPSConfigRestClient.post(dasConfigDetails, processName, processVersion);
+
+            //logging out from DAS Admin Services
```

Comment:
loginServiceClient.logOut() has been called in finally block. We can remove it from here as finally block call it anyway.